src/lib.rs

```diff
@@ -274,11 +274,16 @@ impl Dependencies {
         self.libs.get(name)
     }
 
-    /// An iterator visiting all system dependencies in arbitrary order.
+    /// A vector listing all system dependencies in sorted (for build reproducibility) order.
     /// The first element of the tuple is the name of the `toml` key defining the
     /// dependency in `Cargo.toml`.
-    pub fn iter(&self) -> impl Iterator<Item = (&str, &Library)> {
-        self.libs.iter().map(|(k, v)| (k.as_str(), v))
+    pub fn iter(&self) -> Vec<(&str, &Library)> {
+        let mut v = self.libs
```

Comment:
Couldn't we change `probe_pkg_config()` to sort the libs before storing them? That would prevent us from breaking the API.

Reply:
It's also possible to bring back IterTools and use its sorted_by_key() which produces an Iterator - or if that's not desirable, then somehow figure out how to convert the vector to the iterator (I tried to return v.iter() only to get cryptic errors from rustc - my rust is not that good to understand them).

Reply:
yeah I would have used `itertools` as well but have been asked to keep the deps as minimal as possible so let's do it manually.

Reply:
Can you please take my latest code then and figure out how to tweak it to produce the needed iterator from the sorted vector?

Reply:
I  don't think we can (easily), so let's keep the API change then.